src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let objectCopy = { ...state };
+  const stateCopy = [];
+
+  for (const key in actions) {
+    if (actions[key].type === 'addProperties') {
+      Object.assign(objectCopy, actions[key].extraData);
+      stateCopy.push({ ...objectCopy });
+    }
+
+    if (actions[key].type === 'removeProperties') {
+      for (const rem in actions[key].keysToRemove) {
+        delete objectCopy[actions[key].keysToRemove[rem]];
+      }
+      stateCopy.push({ ...objectCopy });
+    }
+
+    if (actions[key].type === 'clear') {
+      objectCopy = {};
+      stateCopy.push({ ...{} });
```

Comment:
```suggestion
      stateCopy.push({});
```
or 
```suggestion
      stateCopy.push(...objectCopy);
```
And move it after all if blocks (you do the same in each of them), or after `switch` block